wordpress-mobile/WordPress-iOS
# [Tooling] Re-enable localization linting

### What

This re-enables the calls to the `ios_lint_localizations` action, which had been disabled in the past due to the fact that one of our keys were too long and thus came back truncated from GlotPress, which limits the keys to 256 characters.

This has since been addressed by our recent improvements in our Localization Tooling (ref: paaHJt-3bh-p2) which means we are now able to properly fix that problematic string (without having to resort to some workaround in the code like splitting the copy in two separate keys) by using a key that is not the English copy 🎉 

### To Test

 - Run `bundle exec fastlane run ios_lint_localizations input_dir:'WordPress/Resources'` to test the action (recently fixed by https://github.com/wordpress-mobile/release-toolkit/pull/353)

At that point given the state of GlotPress, you should get the following violations printed:
```
[23:16:00]: Inconsistencies found between 'en' and 'nl':

`<strong>Once</strong> a week at %@` expected placeholders for [String] but found [] instead.
`<strong>Twice</strong> a week at %@` expected placeholders for [String] but found [] instead.

[23:16:00]: Inconsistencies found between 'en' and 'ro':

`%1$d Like` expected placeholders for [Int] but found [] instead.
`_You and %1$d blogger_ like this.` expected placeholders for [Int] but found [] instead.

[23:16:00]: Inconsistencies found between 'en' and 'zh-Hans':

`To use this app for %@ you'll need to have the Jetpack plugin installed and activated.` expected placeholders for [String] but found [] instead.


[!] Inconsistencies found during Localization linting. Aborting.
```

Those seem to be genuine violations, that we will need to fix at their root in GlotPress first (or ask Polyglots and +i18n to help us do so) before merging that PR (or at least before we do the next beta after merging that PR).

### PR Status

 - This PR points to a branch of the `release-toolkit`. We need to update the reference in the `Pluginfile` once we do a new release of the toolkit, or at least once we merge https://github.com/wordpress-mobile/release-toolkit/pull/353 to trunk and can point to it.
 - We should address the remaining existing violations by fixing the translations in GlotPress first, before merging this PR, otherwise the linter will (rightfully) start complaining about them on every beta. Hence the "DO NOT MERGE" label.
